crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -267,17 +268,30 @@ impl<'a> Submitter<'a> {
                 name
             );
 
-            let transactions = transactions
-                .into_iter()
+            let transaction_hashes = transactions
+                .iter()
                 .map(|(handle, _)| handle.tx_hash)
                 .collect::<Vec<_>>();
 
             loop {
-                if let Some(receipt) =
-                    find_mined_transaction(&self.contract.raw_instance().web3(), &transactions)
-                        .await
+                if let Some(receipt) = find_mined_transaction(
+                    &self.contract.raw_instance().web3(),
+                    &transaction_hashes,
+                )
+                .await
                 {
-                    tracing::debug!("{} found mined transaction {:?}", name, receipt);
+                    let submitted_by = transactions
+                        .iter()
+                        .find(|(handle, _)| handle.tx_hash == receipt.transaction_hash)
+                        .map(|(handle, _)| handle.submitted_by.as_str())
+                        .unwrap_or("unknown");
+                    tracing::debug!(
+                        "{} found mined transaction {:?} submitted by {}",
+                        name,
+                        receipt,
+                        submitted_by
+                    );
+                    track_mined_transactions(submitted_by);
```

Comment:
I think we could write as well `track_mined_transactions(name)` since this part of the code will be entered only by the strategy that actually mined the transaction? (unless we want to even know which custom_node mined the transaction although I don't see why would we need that...)

Reply:
I thought for each settlement we create a single `Submitter` which knows about all the enabled submission strategies.
But after a closer look it seems that every `Submitter` only represents a single strategy which would make this whole fingerprinting stuff unnecessary (I agree that it's not important which specific custom node was successful). 🤦‍♂️